Tensile/KernelWriterAssembly.py

```diff
@@ -3108,22 +3108,30 @@ def allocateResources(self, kernel):
       kStr += ".if 0\n"
 
     # add offset to buffer
+    tmpOffset = self.sgprPool.checkOutAligned(2, 2, preventOverflow=0)
+
     if not kernel["_GlobalAccumulation"]:
-      kStr += inst("s_lshl_b32", sgpr("OffsetD"), sgpr("OffsetD"), hex(log2(self.bpeCexternal)), "elements offset to bytes offset")
-      kStr += inst("s_add_u32",  sgpr("AddressD+0"), sgpr("AddressD+0"), sgpr("OffsetD"), "add offset to buffer address")
-      kStr += inst("s_addc_u32", sgpr("AddressD+1"), sgpr("AddressD+1"), 0, "add offset to buffer address")
+      kStr += inst("s_mov_b32", sgpr(tmpOffset), sgpr("OffsetD"), "copy to temp b64 to prevent overflow with large offset")
```

Comment:
We should initialize tmpOffset+1 to 0 before using 64bit shift.

Reply:
Thanks! Just posted an update